<h4>Import bibliotek</h4>

In [10]:
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns
import numpy as np

sns.set_style("darkgrid")


<h4>Wczytanie plików do analizy</h4>

In [2]:
qual_df1 = pd.read_excel("Metabolomika_Quality1.xlsx")
qual_df2 = pd.read_excel("Metabolomika_Quality2.xlsx", sheet_name="Arkusz2")

<h4>Filtrowanie blanków do analizy</h4>

In [3]:
qc1 = qual_df1[qual_df1['Group'] == "QC"]

In [4]:
qc1.head()

Id Group            M1            M2            M3            M4  \
198  QC1    QC  4.621374e+10  4.002604e+10  4.385504e+10  4.217619e+10   
199  QC2    QC  4.658385e+10  3.182933e+10  3.620877e+10  4.281087e+10   
200  QC3    QC  4.241968e+10  2.701020e+10  3.021078e+10  4.245005e+10   
201  QC4    QC  2.427638e+10  1.456812e+10  1.444189e+10  1.933829e+10   
202  QC5    QC  4.444659e+10  2.040456e+10  2.482082e+10  3.920081e+10   

               M5            M6            M7            M8  ...  \
198  4.477451e+10  3.394110e+10  1.922884e+10  2.738927e+10  ...   
199  4.279061e+10  3.628560e+10  2.022500e+10  2.322643e+10  ...   
200  4.075320e+10  3.724620e+10  2.057957e+10  2.084970e+10  ...   
201  2.584487e+10  2.250679e+10  1.288211e+10  1.069729e+10  ...   
202  3.949111e+10  4.417547e+10  2.126256e+10  1.851727e+10  ...   

            M1190         M1191         M1192         M1193         M1194  \
198  3.390495e+07  6.690105e+07  2.859299e+07  5.283338e+07  4.605379e+07   
199  1.491863e+07  7.022714e+07  1.029436e+07  5.621284e+07  3.949647e+07   
200  8.604308e+06  7.564514e+07  9.988790e+06  5.459278e+07  3.673037e+07   
201  1.257410e+07  4.168793e+07  1.014177e+07  4.156950e+07  2.888834e+07   
202  7.695319e+06  7.457295e+07  8.589095e+06  5.979793e+07  3.553041e+07   

            M1195         M1196         M1197         M1198         M1199  
198  5.859670e+07  7.065457e+07  5.996072e+07  2.553846e+07  9.809821e+07  
199  6.455666e+07  4.924071e+07  4.800692e+07  4.309458e+07  4.263236e+07  
200  6.281378e+07  4.375875e+07  4.410359e+07  7.321657e+06  7.428908e+07  
201  6.706637e+07  4.329700e+07  3.326763e+07  2.822598e+07  7.216605e+07  
202  6.012362e+07  3.778953e+07  4.472268e+07  2.206086e+07  9.076398e+07  

[5 rows x 1201 columns]

In [5]:
qc2 = qual_df2[qual_df2['Group'] == "QC"]

In [7]:
qc2.head()

ID Group            M1            M2            M3            M4  \
10  QC1    QC  7.850861e+08  6.324966e+10  1.997776e+09  4.126819e+09   
11  QC2    QC  8.443678e+08  6.148879e+10  2.045706e+09  4.279435e+09   
12  QC3    QC  8.159744e+08  5.806665e+10  2.038377e+09  4.203325e+09   
13  QC4    QC  7.892523e+08  5.422457e+10  2.041150e+09  4.111756e+09   
14  QC5    QC  6.641764e+08  5.117223e+10  2.015382e+09  4.169520e+09   

              M5            M6            M7            M8  ...  \
10  2.526125e+09  2.148001e+08  9.551606e+09  2.148001e+08  ...   
11  2.526125e+09  2.288102e+08  9.518574e+09  2.307603e+08  ...   
12  2.526125e+09  2.272386e+08  9.347088e+09  2.285111e+08  ...   
13  2.526125e+09  2.114020e+08  9.059614e+09  2.141895e+08  ...   
14  1.970797e+09  1.697975e+08  7.656014e+09  1.732363e+08  ...   

            M1190          M1191          M1192         M1193          M1194  \
10  905451.913800  540596.952978  786977.366109  9.989060e+05  901019.178715   
11  948422.772642  561572.649209  746218.544403  1.307064e+06  755434.231325   
12  948973.792423  553296.298159  862601.342414  1.253462e+06  731255.688674   
13  899222.959454  570903.459745  775830.745835  1.233514e+06  682619.103833   
14  905451.913800  557780.963139  789292.576319  1.244639e+06  759794.969719   

            M1195          M1196          M1197          M1198          M1199  
10  946119.387993  551958.006014  783966.321469  613834.067917  746490.693947  
11  932361.859760  444072.359865  821735.118365  789253.717846  743652.056938  
12  954391.433374  607936.291864  801509.068786  777302.634066  762055.896851  
13  966908.684026  556818.461665  752031.227315  764311.396698  733697.434773  
14  946119.387993  562540.822613  784387.379206  777302.634066  746490.693947  

[5 rows x 1201 columns]

<h4>Macierz korelacji kowariancji dla blanków</h4>

In [12]:
qc1.iloc[:,2:].T.corr()

198       199       200       201       202       203       204  \
198  1.000000  0.992785  0.981956  0.953082  0.966145  0.960297  0.939607   
199  0.992785  1.000000  0.994187  0.962782  0.986538  0.976023  0.953418   
200  0.981956  0.994187  1.000000  0.964038  0.990728  0.976607  0.958067   
201  0.953082  0.962782  0.964038  1.000000  0.962578  0.958550  0.945006   
202  0.966145  0.986538  0.990728  0.962578  1.000000  0.979054  0.956685   
203  0.960297  0.976023  0.976607  0.958550  0.979054  1.000000  0.984265   
204  0.939607  0.953418  0.958067  0.945006  0.956685  0.984265  1.000000   
205  0.925532  0.937969  0.932768  0.950687  0.941342  0.977584  0.968509   
206  0.948893  0.963087  0.966043  0.953363  0.965052  0.986435  0.969481   
207  0.954997  0.971849  0.973101  0.957272  0.975690  0.998351  0.985751   
208  0.903713  0.921174  0.923937  0.893661  0.925922  0.897815  0.858347   
209  0.901408  0.917500  0.920648  0.893558  0.922042  0.894425  0.857431   
210  0.898208  0.914507  0.916124  0.887786  0.918351  0.896925  0.857117   
211  0.889923  0.903295  0.902035  0.900915  0.908513  0.895279  0.858510   
212  0.898489  0.914911  0.917585  0.890019  0.919511  0.898114  0.859508   
213  0.897806  0.913400  0.915464  0.889309  0.917719  0.895853  0.857536   
214  0.923600  0.936404  0.937804  0.917521  0.935364  0.916697  0.882034   
215  0.927001  0.937157  0.936280  0.912666  0.929535  0.914096  0.877834   
216  0.921966  0.934682  0.935691  0.912759  0.933450  0.917673  0.883240   
217  0.910423  0.923703  0.925290  0.901373  0.923033  0.910035  0.877481   
218  0.918594  0.930039  0.928887  0.908342  0.928124  0.913355  0.882471   
219  0.915092  0.927348  0.928021  0.905624  0.927102  0.910508  0.878168   

          205       206       207  ...       210       211       212  \
198  0.925532  0.948893  0.954997  ...  0.898208  0.889923  0.898489   
199  0.937969  0.963087  0.971849  ...  0.914507  0.903295  0.914911   
200  0.932768  0.966043  0.973101  ...  0.916124  0.902035  0.917585   
201  0.950687  0.953363  0.957272  ...  0.887786  0.900915  0.890019   
202  0.941342  0.965052  0.975690  ...  0.918351  0.908513  0.919511   
203  0.977584  0.986435  0.998351  ...  0.896925  0.895279  0.898114   
204  0.968509  0.969481  0.985751  ...  0.857117  0.858510  0.859508   
205  1.000000  0.959636  0.979510  ...  0.856674  0.883847  0.858531   
206  0.959636  1.000000  0.985372  ...  0.906633  0.901237  0.907143   
207  0.979510  0.985372  1.000000  ...  0.891891  0.891802  0.893251   
208  0.849915  0.908073  0.892336  ...  0.995389  0.973392  0.995462   
209  0.850237  0.903979  0.889633  ...  0.995699  0.981160  0.997694   
210  0.856674  0.906633  0.891891  ...  1.000000  0.981595  0.998404   
211  0.883847  0.901237  0.891802  ...  0.981595  1.000000  0.984194   
212  0.858531  0.907143  0.893251  ...  0.998404  0.984194  1.000000   
213  0.857530  0.904466  0.890629  ...  0.997511  0.984918  0.999187   
214  0.870578  0.927851  0.911268  ...  0.986371  0.965774  0.986507   
215  0.869856  0.924484  0.909071  ...  0.988393  0.967763  0.988288   
216  0.875176  0.927469  0.912684  ...  0.991156  0.972864  0.991768   
217  0.869006  0.920596  0.905114  ...  0.990972  0.972204  0.991057   
218  0.877673  0.922312  0.908687  ...  0.992938  0.977591  0.991678   
219  0.872408  0.919233  0.905681  ...  0.991912  0.977690  0.993025   

          213       214       215       216       217       218       219  
198  0.897806  0.923600  0.927001  0.921966  0.910423  0.918594  0.915092  
199  0.913400  0.936404  0.937157  0.934682  0.923703  0.930039  0.927348  
200  0.915464  0.937804  0.936280  0.935691  0.925290  0.928887  0.928021  
201  0.889309  0.917521  0.912666  0.912759  0.901373  0.908342  0.905624  
202  0.917719  0.935364  0.929535  0.933450  0.923033  0.928124  0.927102  
203  0.895853  0.916697  0.914096  0.917673  0.910035  0.913355  0.910508  
204  0.857536  0.882034  0.877834  0.883240  0.87

In [13]:
np.min(qc1.iloc[:,2:].T.corr())

np.float64(0.8499146062622087)

In [9]:
qc2.iloc[:,2:].T.corr()

10        11        12        13        14        15        16  \
10  1.000000  0.999913  0.999689  0.999316  0.999550  0.999832  0.999457   
11  0.999913  1.000000  0.999900  0.999656  0.999591  0.999949  0.999726   
12  0.999689  0.999900  1.000000  0.999887  0.999482  0.999892  0.999914   
13  0.999316  0.999656  0.999887  1.000000  0.999177  0.999636  0.999908   
14  0.999550  0.999591  0.999482  0.999177  1.000000  0.999576  0.999406   
15  0.999832  0.999949  0.999892  0.999636  0.999576  1.000000  0.999697   
16  0.999457  0.999726  0.999914  0.999908  0.999406  0.999697  1.000000   
17  0.999622  0.999860  0.999978  0.999904  0.999360  0.999840  0.999902   
18  0.999737  0.999922  0.999981  0.999886  0.999478  0.999885  0.999890   
19  0.999572  0.999804  0.999957  0.999926  0.999366  0.999784  0.999919   
20  0.999680  0.999834  0.999915  0.999773  0.999261  0.999813  0.999771   
21  0.998768  0.998605  0.998387  0.998048  0.997886  0.998419  0.998196   
22  0.999827  0.999777  0.999655  0.999313  0.999376  0.999671  0.999446   
23  0.999427  0.999679  0.999847  0.999785  0.999183  0.999669  0.999840   
24  0.999782  0.999912  0.999872  0.999676  0.999410  0.999930  0.999686   
25  0.999781  0.999907  0.999901  0.999709  0.999444  0.999904  0.999764   
26  0.999748  0.999905  0.999953  0.999844  0.999455  0.999861  0.999850   
27  0.999218  0.999538  0.999815  0.999806  0.999027  0.999542  0.999799   

          17        18        19        20        21        22        23  \
10  0.999622  0.999737  0.999572  0.999680  0.998768  0.999827  0.999427   
11  0.999860  0.999922  0.999804  0.999834  0.998605  0.999777  0.999679   
12  0.999978  0.999981  0.999957  0.999915  0.998387  0.999655  0.999847   
13  0.999904  0.999886  0.999926  0.999773  0.998048  0.999313  0.999785   
14  0.999360  0.999478  0.999366  0.999261  0.997886  0.999376  0.999183   
15  0.999840  0.999885  0.999784  0.999813  0.998419  0.999671  0.999669   
16  0.999902  0.999890  0.999919  0.999771  0.998196  0.999446  0.999840   
17  1.000000  0.999962  0.999934  0.999925  0.998303  0.999596  0.999843   
18  0.999962  1.000000  0.999955  0.999901  0.998484  0.999671  0.999803   
19  0.999934  0.999955  1.000000  0.999858  0.998447  0.999606  0.999841   
20  0.999925  0.999901  0.999858  1.000000  0.998337  0.999684  0.999823   
21  0.998303  0.998484  0.998447  0.998337  1.000000  0.998846  0.997871   
22  0.999596  0.999671  0.999606  0.999684  0.998846  1.000000  0.999483   
23  0.999843  0.999803  0.999841  0.999823  0.997871  0.999483  1.000000   
24  0.999830  0.999885  0.999778  0.999784  0.998404  0.999578  0.999585   
25  0.999877  0.999909  0.999823  0.999845  0.998353  0.999641  0.999714   
26  0.999952  0.999958  0.999917  0.999902  0.998549  0.999707  0.999779   
27  0.999844  0.999768  0.999831  0.999811  0.997891  0.999342  0.999752   

          24        25        26        27  
10  0.999782  0.999781  0.999748  0.999218  
11  0.999912  0.999907  0.999905  0.999538  
12  0.999872  0.999901  0.999953  0.999815  
13  0.999676  0.999709  0.999844  0.999806  
14  0.999410  0.999444  0.999455  0.999027  
15  0.999930  0.999904  0.999861  0.999542  
16  0.999686  0.999764  0.999850  0.999799  
17  0.999830  0.999877  0.999952  0.999844  
18  0.999885  0.999909  0.999958  0.999768  
19  0.999778  0.999823  0.999917  0.999831  
20  0.999784  0.999845  0.999902  0.999811  
21  0.998404  0.998353  0.998549  0.997891  
22  0.999578  0.999641  0.999707  0.999342  
23  0.999585  0.999714  0.999779  0.999752  
24  1.000000  0.999931  0.999845  0.999512  
25  0.999931  1.000000  0.999846  0.999637  
26  0.999845  0.999846  1.000000  0.999704  
27  0.999512  0.999637  0.999704  1.000000

In [14]:
np.min(qc2.iloc[:,2:].T.corr())

np.float64(0.997871487716682)

W idealnym przypadku wszystkie wartości macierzy korelacji powinne być równe albo chociaż bliskie 1. 

W próbkach kontroli jakości z pliku nr 1 te wartości są w znacznej części przypadków bliskie 1. z najniższą wartością równą $\sim 0.850$.

W blankach z pliku nr 2 te wartości są praktycznie idealne ze względu na policzone wartości korelacji z najniższą wartością $\sim 0.998$.